node/silkworm/rpc/call.hpp

```diff
@@ -479,12 +489,16 @@ class BidirectionalStreamingRpc : public BaseRpc {
         SILK_TRACE << "BidirectionalStreamingRpc::BidirectionalStreamingRpc END new request issued [" << this << "]";
     }
 
+    virtual void start() = 0;
+
     //! Hook called when a new incoming request from some client has come in for this RPC.
     /// For client-streaming and bidirectional streaming RPCs, a request from client can come in multiple times so
     /// \ref process() may be called repeatedly.
     /// @param request the incoming request or nullptr to indicate end-of-stream for client stream
     virtual void process(const Request* request) = 0;
 
+    virtual void end() = 0;
+
     bool send_response(const Response& response) {
         response_queue_.push_back(std::move(response));
```

Comment:
move() here is an oversight or you need to pass Response by copy.

Reply:
Oops, you are right, thanks for reporting.

The concrete RPC classes call the `send_response` method typically in two different use cases:
- either passing a reference to a state variable having the same lifetime as the RPC instance
- or passing a stack-allocated `Response` object after it has been filled with data

So a wise solution here could be introduce overloading and:
- getting rid of the `std::move` call in this version taking `const Response&`
- adding another version taking `Response&&` as formal parameter